File: Integrais - unidade 3/TrapezioComposto.java
import java.util.Scanner;

public class TrapezioComposto {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        double a, b;
        System.out.println("Digite o valor de a: ");
        a = sc.nextDouble();
        System.out.println("Digite o valor de b: ");
        b = sc.nextDouble();
        sc.close();

        double n = b - a;
        double h = (b - a) / n;

        double soma = f(a) + f(b);
        for (int i = 1; i < n; i++) {
            soma += 2 * f(a + i * h);
        }

        System.out.println("A área aproximada é: " + (h / 2) * soma);

    }

    public static double f(double x) {
        return -(x * x) + (3 * x);
    }
}

File: Integrais - unidade 3/TrapezioComposto.py
def f(x):
    return -(2*x) - 3

def trapezio_composto(a, b, n):
    h = (b - a) / n
    soma = f(a) + f(b)
    for i in range(1, n):
        soma += 2 * f(a + i * h)
    return (h / 2) * soma

def main():
    a = float(input("Digite o valor de a: "))
    b = float(input("Digite o valor de b: "))
    n = int(input("Digite o valor de n: "))

    area_aproximada = trapezio_composto(a, b, n)
    print("A área aproximada é:", area_aproximada)

if __name__ == "__main__":
    main()
